### ISSUE TITLE
Threading API

### ISSUE THREAD

**Issue by benparsons** [matrix-org/matrix-doc#1198]
Documentation: https://docs.google.com/document/d/1bLAcYBvTYp2XNvUG-DuYv4E0uWThz_Cr6PHzspq7e60/edit
Author: Half-Shot

**Comment by erlend-sh** [comment]
Superseded by the official RFC? #2326

**Comment by ara4n** [comment]
nope, MSC #2326 is just for filtering by labels rather than navigating a tree of threads.

**Comment by ara4n** [comment]
may be superseded by #2836 tho (if it ends up including swimlanes via #2326)
